### PR TITLE
GoMod: Allow for analyzing modules which are part of a multi-module workspace

### PR DESCRIPTION
**Pull request by fviernau**
See individual commits.

Fixes: #6979.

### CODE REVIEW

File: analyzer/src/funTest/assets/projects/synthetic/gomod-workspaces-main-module-expected-output.yml [hunk 1]
```diff
@@ -1,25 +1,85 @@
 ---
```
Comment (MarcelBochtler) [inline analyzer/src/funTest/assets/projects/synthetic/gomod-workspaces-main-module-expected-output.yml:1]: Commit message nit: `[1 ]` should be [1]

File: analyzer/src/funTest/assets/projects/synthetic/gomod-workspaces/go.work [hunk 2]
```diff
@@ -0,0 +1,6 @@
+go 1.18
+
+use (
+    .
+	other-module
+)
```
Comment (sschuberth) [inline analyzer/src/funTest/assets/projects/synthetic/gomod-workspaces/go.work:6]: Nit: Missing final newline.

File: analyzer/src/funTest/assets/projects/synthetic/gomod-workspaces/go.work [hunk 3]
```diff
@@ -0,0 +1,6 @@
+go 1.18
+
+use (
+    .
+	other-module
```
Comment (sschuberth) [inline analyzer/src/funTest/assets/projects/synthetic/gomod-workspaces/go.work:5]: Nit: These lines use different indentation.

File: analyzer/src/funTest/assets/projects/synthetic/gomod-workspaces/other-module/main.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package main
+
+import (
+    "fmt"
+	"strings"
```
Comment (sschuberth) [inline analyzer/src/funTest/assets/projects/synthetic/gomod-workspaces/other-module/main.go:5]: Nit: These lines use different indentation.

File: analyzer/src/main/kotlin/managers/GoMod.kt [hunk 5]
```diff
@@ -87,7 +87,7 @@ class GoMod(
     private val environment by lazy {
         mapOf(
             "GOPATH" to goPath.absolutePath,
-            "GOPROXY" to "direct",
+            "GOPROXY" to "direct"
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/GoMod.kt]: This should go to the previous commit.

File: analyzer/src/funTest/kotlin/managers/GoModFunTest.kt [hunk 6]
```diff
@@ -57,23 +57,21 @@ class GoModFunTest : StringSpec({
...
 
     "Project dependencies of a main module in a multi-module workspaces are detected correctly" {
-        // TODO: The expected result illustrates the issue: https://github.com/oss-review-toolkit/ort/issues/6979.
         val definitionFile = testDir.resolve("gomod-workspaces/go.mod")
         val expectedResultFile = testDir.resolve("gomod-workspaces-main-module-expected-output.yml")
 
         val result = create("GoMod").resolveSingleProject(definitionFile)
 
-        result.withInvariantIssues().toYaml() should matchExpectedResult(expectedResultFile, definitionFile)
+        result.toYaml() should matchExpectedResult(expectedResultFile, definitionFile)
```
Comment (sschuberth) [inline analyzer/src/funTest/kotlin/managers/GoModFunTest.kt:65]: Should `withInvariantIssues()` already have been dropped before, in the commit that introduces this? Same below.

**Review by sschuberth — CHANGES_REQUESTED** [review]

**Review by sschuberth — APPROVED** [review]

**Review by MarcelBochtler — APPROVED** [review]

**Pull request merged**